fdbserver/BlobWorker.actor.cpp

```diff
@@ -73,351 +74,101 @@
  * It is responsible for managing the change feeds for those granules, and for consuming the mutations from
  * those change feeds and writing them out as files to blob storage.
  */
-
-struct GranuleStartState {
-	UID granuleID;
-	Version changeFeedStartVersion;
-	Version previousDurableVersion;
-	Optional<std::pair<KeyRange, UID>> splitParentGranule;
-	bool doSnapshot;
-	std::vector<GranuleFiles> blobFilesToSnapshot;
-	Optional<GranuleFiles> existingFiles;
-	Optional<GranuleHistory> history;
-};
-
-// TODO: add more (blob file request cost, in-memory mutations vs blob delta file, etc...)
-struct GranuleReadStats {
-	int64_t deltaBytesRead;
-
-	void reset() { deltaBytesRead = 0; }
-
-	GranuleReadStats() { reset(); }
-};
-
-struct GranuleMetadata : NonCopyable, ReferenceCounted<GranuleMetadata> {
```

Comment:
An alternate would be move the declaration of the struct into a common header file, may be BlobWorkerCommon.h and leave the definition in the BlobWorker.cpp; not sure if BlobServerCommon.h is the right place ?

Reply:
BlobWorkerCommon.h is in the fdbclient folder, i think both BlobWorkerData and GranuleMetadata should be in the fdb server folder

Reply:
I can understand the confusion, fdbclient files can be included from both fdbserver and/or fdbclient, however, other way around is not true.

Given BlobWorker.cpp is defined in fdbserver, the reason BlobWorkerCommon is defined in fdbclient is because, the file is included from BlobGranuleReader (fdbclient module).

Reply:
Also, I noticed there is `BlobGranuleCommon.h` seems like GranuleMetadata may fit better there?

Reply:
GranuleMetadata will use some structs defined inside BlobGranuleSeverCommon.actor.h e.g struct GranuleFiles, which can not be included by fdbclient file